proposals/plugins.md

```diff
@@ -0,0 +1,146 @@
+The deployment can be performed either via Kubernetes YAML or Helm Chart both of which would be a file/directory reference.
+
+### Additional Worker IAM Permissions / kube2iam
+
+Additional worker IAM permissions requires either additions to the existing IAM policies or new IAM policies
+being attached to the worker instance role. This could be done in a few ways, for example `initialise` could
+output the additional CloudFormation template snippet needed along with name/ref of the IAM policy.
+The policies and policy document for the workers would be primarily managed by a core plugin that deals
+with the worker permissions. It would take the output of `initialise` and render the final CloudFormation required.
+
+The kube2iam add-on requires Kubernetes YAML or a Helm Chart to be installed which is covered by the Dex
+example above plus changes to the worker IAM permissions.
+
+### Bastion Host
+
+A bastion host requires additional CloudFormation to deploy the instance into the correct part of the
+VPC. It therefore requires knowledge of the VPC/network setup as input. This plugin would take the 
+configuration in `initialise` and output the additional CloudFormation based on the VPC settings.
+
+### CloudFormation Renderer
+
+During the `initialise` stage a core CloudFormation render plugin could output the basic CloudFormation templates for the cluster. Then during the `render` stage the same core plugin could choose to take all the output across all plugins from `initialise` and render the final CloudFormation.
+
+### Kube API Server
+
+A Kubernetes API server plugin could be responsible for main piece of cloud config that sets up the API server. It could then also perform additional validation during the `render` stage to ensure any plugins that have added to the kube API server parameters primary construct have not made the final state invalid.
+
+# Implementation
+
+* [Helm](https://github.com/kubernetes/helm) should be the main method for deploying additional components into the Kubernetes cluster
+* How can kube-aws deal with plugin ordering? The ordering could be defined in a fixed YAML file and anything not present is always run last.
+* What is the process to include and compile plugins?
+   * One idea is to use [go-plugin](https://github.com/hashicorp/go-plugin) since the native 1.8 plugins is not mature yet
+* We need to protect sensitive information if uploading files, some Helm charts have secrets as config values so how will these be input?
+
+## Configuration
+
+All plugins will be enabled and configured via the `cluster.yaml` plus a set of plugin specific
+configuration files. However, as above, the whole configuration model will be passed to all plugins
+so they can access parameters from another plugin if necessary.
+
+## Plugin Locations
+
+* Built-in plugins can either be part of the kube-aws binary or downloadable from a secure source 
+if/when the user enables that plugin. The second is recommended.
```

Comment:
@c-knowles Thanks for the confirmation 👍 
I'm even considering that a *built-in* plugin, when it is enabled, should be extracted from the kube-aws binary and then *rendered* into the working tree like *adhoc* plugins, so that it can be customized freely by the user without rebuilding the whole kube-aws binary.
What do you think?

cc @redbaron